pycqed/measurement/measurement_control.py

```diff
@@ -386,7 +453,11 @@ def measure_soft_adaptive(self, method=None):
 
     @Timer()
     def measure_hard(self):
+        # Tell the detector_function to call print_progress for intermediate
+        # progress reports during get_detector_function.values.
+        self.detector_function.progress_callback = self.print_progress
         new_data = np.array(self.detector_function.get_values()).T
+        self.detector_function.progress_callback = None
```

Comment:
Why do you need to set it back to None?

Reply:
Just for the sake of cleaning up. The structure of pycqed would in principle allow to use a detector function with various measurement control objects. Thus, MC should not leave anything that is specific to a particular instance of MC in the detector function.

Reply:
👍